### ISSUE TITLE
How to inject dependencies to custom redefined Client Class

### ISSUE THREAD

**Issue by nakashu** [8p/EightPointsGuzzleBundle#234]
| Q                | A
| ---------------- | -----
| Symfony version  | 4.1.4
| Bundle version   | 7.4.0

How to inject more dependencies into redefined Client Class as defined here:
- https://github.com/8p/EightPointsGuzzleBundle/blob/master/src/Resources/doc/redefine-client-class.md
- https://github.com/8p/EightPointsGuzzleBundle/blob/master/src/Resources/doc/autowiring-clients.md

As the service is build dynamically in EightPointGuzzleExtension.php it seems at least to me, that its not possible to simply add more dependencies.

Some uses cases would be %kernel.default_locale%, or other helper services.

I also tried using the options key, and getting it thru config, but this would not allowed as unrecognized option. thanks for any feedback

**Comment by gregurco** [comment]
Hello @nakashu 
Why do you need to redefine client class and to write there some logic? I think it's a little bit wring way. Why not to define request/response interceptor? See how it works here: https://github.com/8p/EightPointsGuzzleBundle/blob/master/src/Resources/doc/intercept-request-and-response.md
Or do you have some specific requirement?

**Comment by nakashu** [comment]
@gregurco thanks for the response.

It just seems to much for adding a simple requirement or parameter like the mentioned locale to create a listener, just my point of view.

So basically just because its just such a small thing I would like to DI it into the class instead of adding it to listener. Thus making it easier for others to understand whats happening without splitting up the code to much.

In more general use case, it seems right that an api client would need some configurable parameters and/or services (for instance a custom serializer that would be injected), and for me it seems its a bit too much hassle to implement a custom listener. Yes listener are great for separation of concerns, but for a simple parameter to have it accessible seems to much.

So basically, the question still stands if it currently possible. thanks.

**Comment by gregurco** [comment]
Probably it's possible to do without any classes and just with configuration. You gave as example the locale. How will it be used in the request? As parameter in body of all requests or how?

**Comment by nakashu** [comment]
it would be in the logic of many but not all requests, in some request method.

```php 
public function getStuff($criteria = [], $locale = NULL) 
{
    if (NULL = $locale) {
        $locale = $this->default_locale // or how ever to get it here, from container
    }
    return $this->get('/stuff', ['locale' => $locale])
}
```
